`startOfflineExecutor` is not compatible with `queryCollection` types

- [x] I've validated the bug against the latest version of DB packages

**Describe the bug**
When a `queryCollection` is added in the collections field of `startOfflineExecutor`, a TypeScript error is produced.

**To Reproduce**
- Stackblitz repro: https://stackblitz.com/edit/vitejs-vite-7ityyqnr?file=package.json
- Run `pnpm typecheck` in the terminal or hover over line 25 in `main.ts`.
**Expected behavior**
There should not be a TypeScript error when using the latest query collection package with offline transactions.

**Error Message**
```
❯ pnpm typecheck

> vite-starter@0.0.0 typecheck /home/projects/vitejs-vite-7ityyqnr
> tsc

src/main.ts:25:18 - error TS2322: Type 'Collection<{ id: string; }, string, UtilsRecord & QueryCollectionUtils<{ id: string; }, string, { id: string; }, unknown>, ZodObject<{ id: ZodString; }, $strip>, { ...; }> & NonSingleResult' is not assignable to type 'Collection<Record<string, unknown>, string | number, UtilsRecord, StandardSchemaV1<unknown, unknown>, Record<string, unknown>>'.
  The types of 'config.sync.sync' are incompatible between these types.
    Type '(params: { collection: Collection<{ id: string; }, string, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<{ id: string; }, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }) => void | ... 1 more ... | SyncConfigRes' is not assignable to type '(params: { collection: Collection<Record<string, unknown>, string | number, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<Record<string, unknown>, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }) => void | ... 1 more ... | SyncConfigRes'.
      Types of parameters 'params' and 'params' are incompatible.
        Type '{ collection: Collection<Record<string, unknown>, string | number, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<Record<string, unknown>, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }' is not assignable to type '{ collection: Collection<{ id: string; }, string, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<{ id: string; }, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }'.
          The types of 'collection.config.sync.sync' are incompatible between these types.
            Type '(params: { collection: Collection<Record<string, unknown>, string | number, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<Record<string, unknown>, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }) => void | ... 1 more ... | SyncConfigRes' is not assignable to type '(params: { collection: Collection<{ id: string; }, string, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<{ id: string; }, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }) => void | ... 1 more ... | SyncConfigRes'.
              Types of parameters 'params' and 'params' are incompatible.
                Type '{ collection: Collection<{ id: string; }, string, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<{ id: string; }, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }' is not assignable to type '{ collection: Collection<Record<string, unknown>, string | number, any, any, any>; begin: () => void; write: (message: Omit<ChangeMessage<Record<string, unknown>, string | number>, "key">) => void; commit: () => void; markReady: () => void; truncate: () => void; }'.
                  The types of 'collection.config.getKey' are incompatible between these types.
                    Type '(item: { id: string; }) => string' is not assignable to type '(item: Record<string, unknown>) => string | number'.
                      Types of parameters 'item' and 'item' are incompatible.
                        Property 'id' is missing in type 'Record<string, unknown>' but required in type '{ id: string; }'.

25   collections: { subjects: todosCollection },
                    ~~~~~~~~

  src/main.ts:19:7
    19       id: z.string(),
             ~~~~~~~~~~~~~~
    'id' is declared here.


Found 1 error in src/main.ts:25
```

--- a/packages/offline-transactions/src/outbox/OutboxManager.ts
+++ b/packages/offline-transactions/src/outbox/OutboxManager.ts
@@ -10,7 +10,8 @@ export class OutboxManager {
 
   constructor(
     storage: StorageAdapter,
-    collections: Record<string, Collection>
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    collections: Record<string, Collection<any, any, any, any, any>>
   ) {
     this.storage = storage
     this.serializer = new TransactionSerializer(collections)

--- a/packages/offline-transactions/src/outbox/TransactionSerializer.ts
+++ b/packages/offline-transactions/src/outbox/TransactionSerializer.ts
@@ -7,10 +7,14 @@ import type {
 import type { Collection, PendingMutation } from "@tanstack/db"
 
 export class TransactionSerializer {
-  private collections: Record<string, Collection>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private collections: Record<string, Collection<any, any, any, any, any>>
   private collectionIdToKey: Map<string, string>
 
-  constructor(collections: Record<string, Collection>) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(
+    collections: Record<string, Collection<any, any, any, any, any>>
+  ) {
     this.collections = collections
     // Create reverse lookup from collection.id to registry key
     this.collectionIdToKey = new Map()

--- a/packages/offline-transactions/src/types.ts
+++ b/packages/offline-transactions/src/types.ts
@@ -88,7 +88,8 @@ export interface StorageDiagnostic {
 }
 
 export interface OfflineConfig {
-  collections: Record<string, Collection>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  collections: Record<string, Collection<any, any, any, any, any>>
   mutationFns: Record<string, OfflineMutationFn>
   storage?: StorageAdapter
   maxConcurrency?: number

